code/modules/projectiles/guns/energy.dm

```diff
@@ -69,12 +79,11 @@
 	return ..()
 
 /obj/item/gun/energy/process()
-	if(selfcharge && cell && cell.percent() < 100)
+	if(cell && cell.chargerate && cell.percent() < 100)
 		charge_tick++
 		if(charge_tick < charge_delay)
 			return
 		charge_tick = 0
```

Comment:
```suggestion
	if(cell?.self_recharge)
```

Since we are removing energy guns' self-recharging capability, we can leave only the corresponding part of `/obj/item/gun/energy/process()` behind, which is self-`recharge_newshot`ing as needed. (`on_process()` at cells to make them recharge when they are in a gun isn't a wise choice, so we should do this.)